pkg/runtime/field_export_reconciler.go

```diff
@@ -296,6 +296,9 @@ func (r *fieldExportReconciler) writeToConfigMap(
 ) error {
 	// Construct the data key
 	key := fmt.Sprintf("%s.%s", desired.Namespace, desired.Name)
+	if desired.Spec.To.Key != nil {
+		key = *desired.Spec.To.Key
```

Comment:
What happens if the user enters an empty string?

Reply:
I have changed the logic to consider empty strings. With an empty string value, the key will fall back to `<namespace>.<FieldExport-resource-name>`. a974e46b904f7299f756fcd0a90d5a66522859fb